File: golang/tests/pkg/integration/agentbay_get_integration_test.go
package integration

import (
	"fmt"
	"os"
	"testing"

	"github.com/aliyun/wuying-agentbay-sdk/golang/pkg/agentbay"
)

func TestAgentBayGetAPI(t *testing.T) {
	apiKey := os.Getenv("AGENTBAY_API_KEY")
	if apiKey == "" {
		t.Fatal("AGENTBAY_API_KEY environment variable is not set")
	}

	client, err := agentbay.NewAgentBay(apiKey, nil)
	if err != nil {
		t.Fatalf("Failed to initialize AgentBay client: %v", err)
	}

	fmt.Println("Creating a new session for Get API testing...")
	createResult, err := client.Create(nil)
	if err != nil {
		t.Fatalf("Failed to create session: %v", err)
	}
	createdSession := createResult.Session
	sessionId := createdSession.SessionID
	t.Logf("Session created with ID: %s", sessionId)

	fmt.Println("Testing Get API...")
	result, err := client.Get(sessionId)
	if err != nil {
		t.Fatalf("Failed to get session: %v", err)
	}

	if result == nil {
		t.Fatal("Get returned nil result")
	}

	if !result.Success {
		t.Fatalf("Get failed: %s", result.ErrorMessage)
	}

	if result.Session == nil {
		t.Fatal("Result.Session is nil")
	}

	if result.Session.SessionID != sessionId {
		t.Errorf("Expected SessionID %s, got %s", sessionId, result.Session.SessionID)
	}

	if result.Session.AgentBay == nil {
		t.Error("Session AgentBay reference is nil")
	}

	if result.RequestID == "" {
		t.Error("RequestID should not be empty")
	}

	t.Logf("Successfully retrieved session with ID: %s", result.Session.SessionID)
	t.Logf("Request ID: %s", result.RequestID)

	fmt.Println("Get API test passed successfully")

	fmt.Println("Cleaning up: Deleting the session...")
	deleteResult, err := result.Session.Delete()
	if err != nil {
		t.Fatalf("Failed to delete session: %v", err)
	}
	t.Logf("Session %s deleted successfully", sessionId)
	if !deleteResult.Success {
		t.Error("Session deletion failed")
	}
}

func TestAgentBayGetNonExistentSession(t *testing.T) {
	apiKey := os.Getenv("AGENTBAY_API_KEY")
	if apiKey == "" {
		t.Fatal("AGENTBAY_API_KEY environment variable is not set")
	}

	client, err := agentbay.NewAgentBay(apiKey, nil)
	if err != nil {
		t.Fatalf("Failed to initialize AgentBay client: %v", err)
	}

	fmt.Println("Testing Get API with non-existent session ID...")
	nonExistentSessionId := "session-nonexistent-12345"
	result, err := client.Get(nonExistentSessionId)

	// Get should not return error, but result.Success should be false
	if err != nil {
		t.Fatalf("Unexpected error: %v", err)
	}

	if result.Success {
		t.Fatal("Expected failure for non-existent session, got success")
	}

	if result.ErrorMessage == "" {
		t.Error("ErrorMessage should not be empty for failed result")
	}

	t.Logf("Correctly received error for non-existent session: %s", result.ErrorMessage)
	fmt.Println("Get API non-existent session test passed successfully")
}

func TestAgentBayGetEmptySessionId(t *testing.T) {
	apiKey := os.Getenv("AGENTBAY_API_KEY")
	if apiKey == "" {
		t.Fatal("AGENTBAY_API_KEY environment variable is not set")
	}

	client, err := agentbay.NewAgentBay(apiKey, nil)
	if err != nil {
		t.Fatalf("Failed to initialize AgentBay client: %v", err)
	}

	fmt.Println("Testing Get API with empty session ID...")
	result, err := client.Get("")

	// Get should not return error, but result.Success should be false
	if err != nil {
		t.Fatalf("Unexpected error: %v", err)
	}

	if result.Success {
		t.Fatal("Expected failure for empty session ID, got success")
	}

	expectedErrMsg := "session_id is required"
	if result.ErrorMessage != expectedErrMsg {
		t.Errorf("Expected error message '%s', got '%s'", expectedErrMsg, result.ErrorMessage)
	}

	t.Logf("Correctly received error for empty session ID: %s", result.ErrorMessage)
	fmt.Println("Get API empty session ID test passed successfully")
}

File: typescript/tests/integration/agent-bay-get.integration.test.ts
import { AgentBay } from "../../src/agent-bay";
import { Session } from "../../src/session";
import { log } from "../../src/utils/logger";

describe("AgentBay.get integration tests", () => {
  let agentBay: AgentBay;
  let sessionId: string | undefined;
  let sessionCreated = false;

  beforeAll(async () => {
    const apiKey = process.env.AGENTBAY_API_KEY;
    if (!apiKey) {
      throw new Error("AGENTBAY_API_KEY environment variable is not set");
    }
    agentBay = new AgentBay({ apiKey });

    try {
      log("🚀 Creating a new session for Get API testing...");
      // Create session
      const createResult = await agentBay.create({ imageId: "linux_latest" });
      if (createResult.success && createResult.session) {
        sessionId = createResult.session.sessionId;
        sessionCreated = true;
        log(`✅ Session created with ID: ${sessionId}`);
      } else {
        log("⚠️  Session creation failed, some tests will be skipped");
      }
    } catch (error) {
      log(`❌ Session creation error: ${error}`);
    }
  }); // 60 second timeout for session creation

  test("should retrieve session using Get API", async () => {
    if (!sessionCreated || !sessionId) {
      log("⊘ Skipping test: session was not created");
      return;
    }

    log("📋 Test: Retrieve session using Get API");
    const result = await agentBay.get(sessionId);

    expect(result).toBeDefined();
    expect(result.success).toBe(true);
    expect(result.requestId).toBeDefined();
    expect(result.session).toBeDefined();
    expect(result.session).toBeInstanceOf(Session);
    if (result.session) {
      expect(result.session.sessionId).toBe(sessionId);
      log(`  ✓ Session ID: ${result.session.sessionId}`);
    }
    log(`  ✓ Request ID: ${result.requestId}`);
    log("  ✓ All assertions passed");
  }); // 30 second timeout

  test("should return error for non-existent session", async () => {
    log("📋 Test: Non-existent session error handling");
    const nonExistentSessionId = "session-nonexistent-12345";

    const result = await agentBay.get(nonExistentSessionId);

    expect(result).toBeDefined();
    expect(result.success).toBe(false);
    expect(result.errorMessage).toContain("Failed to get session");

    log(`  ✓ Error correctly received: ${result.errorMessage}`);
  }); // 30 second timeout

  test("should return error for empty session ID", async () => {
    log("📋 Test: Empty session ID error handling");
    const result = await agentBay.get("");

    expect(result).toBeDefined();
    expect(result.success).toBe(false);
    expect(result.errorMessage).toContain("session_id is required");

    log(`  ✓ Error correctly received: ${result.errorMessage}`);
  });

  test("should return error for whitespace session ID", async () => {
    log("📋 Test: Whitespace session ID error handling");
    const result = await agentBay.get("   ");

    expect(result).toBeDefined();
    expect(result.success).toBe(false);
    expect(result.errorMessage).toContain("session_id is required");

    log(`  ✓ Error correctly received: ${result.errorMessage}`);
  });

  afterAll(async () => {
    if (!sessionCreated || !sessionId) {
      log("⊘ No session to clean up");
      return;
    }

    log("🧹 Cleaning up: Deleting the session...");
    try {
      const result = await agentBay.get(sessionId); // Retrieve session object
      if (result.success && result.session) {
        const deleteResult = await result.session.delete();
        if (deleteResult.success) {
          log(`✅ Session ${sessionId} deleted successfully`);
        } else {
          log(`⚠️  Failed to delete session: ${deleteResult.errorMessage}`);
        }
      }
    } catch (error) {
      log(`⚠️  Failed to clean up session: ${error}`);
    }
  }); // 30 second timeout
});
